File: C# .Net/JDI UI Framework/JDI/Web/Selenium/Elements/Complex/CheckList.cs
using Epam.JDI.Core.Interfaces.Complex;
using System;
using OpenQA.Selenium;

namespace Epam.JDI.Web.Selenium.Elements.Complex
{
    public class CheckList : CheckList<IConvertible>, ICheckList
    {
        public CheckList() { }
        public CheckList(By optionsNamesLocator, By allOptionsNamesLocator) 
            : base(optionsNamesLocator, allOptionsNamesLocator) { }
    }
    public class CheckList<TEnum> : MultiSelector<TEnum>, ICheckList<TEnum>
        where TEnum : IConvertible
    {
        public CheckList() { }
        public CheckList(By optionsNamesLocator, By allOptionsNamesLocator) : base(optionsNamesLocator, allOptionsNamesLocator) { }
    }
}

File: jdi-uitest-web/src/main/java/com/epam/jdi/uitests/web/selenium/elements/complex/CheckList.java
package com.epam.jdi.uitests.web.selenium.elements.complex;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */


import com.epam.jdi.uitests.core.interfaces.complex.ICheckList;
import org.openqa.selenium.By;

/**
 * Select control implementation
 *
 * @author Alexeenko Yan
 * @author Belousov Andrey
 */
public class CheckList<TEnum extends Enum> extends MultiSelector<TEnum> implements ICheckList<TEnum> {
    public CheckList() {
        super();
    }

    public CheckList(By optionsNamesLocator) {
        super(optionsNamesLocator);
    }

    public CheckList(By optionsNamesLocator, By allOptionsNamesLocator) {
        super(optionsNamesLocator, allOptionsNamesLocator);
    }
}